### PR TITLE
【PPMix No.4】POINTS-Qwen-2-5-7B-Chat推理对齐

### PR DESCRIPTION
**Pull request by zhaop-l**
[POINTS-Qwen-2-5-7B-Chat](https://huggingface.co/WePOINTS/POINTS-Qwen-2-5-7B-Chat)推理对齐。
image

https://github.com/PaddlePaddle/PaddleMIX/issues/1046#issuecomment-2751017056

### CODE REVIEW

File: paddlemix/examples/points_qwen2_5/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+## 2 环境准备
+1）[安装PaddlePaddle](https://github.com/PaddlePaddle/PaddleMIX?tab=readme-ov-file#3-%EF%B8%8F%E5%AE%89%E8%A3%85paddlepaddle)
+- **python >= 3.10**
+- **paddlepaddle-gpu 要求是3.0.0b2或develop版本**
+```bash
+# 提供三种 PaddlePaddle 安装命令示例，也可参考PaddleMIX主页的安装教程进行安装
+
+# 3.0.0b2版本安装示例 (CUDA 11.8)
+python -m pip install paddlepaddle-gpu==3.0.0b2 -i https://www.paddlepaddle.org.cn/packages/stable/cu118/
```
Comment (nemonameless) [inline paddlemix/examples/points_qwen2_5/README.md:28]: 建议升级到 3.0.0b4也试试

File: paddlemix/examples/points_qwen2_5/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+> 注：
+* 请确保安装了以上依赖，否则无法运行。同时，需要安装 paddlemix/external_ops 下的自定义OP, `python setup.py install`。如果安装后仍然找不到算子，需要额外设置PYTHONPATH
+* (默认开启flash_attn)使用flash_attn 要求A100/A800显卡或者H20显卡。V100请用float16推理。
+
+## 3 快速开始
+
+### 推理
+```bash
+# 单图推理
+python paddlemix/examples/points_qwen2_5/image_infer.py
```
Comment (nemonameless) [inline paddlemix/examples/points_qwen2_5/README.md]: 建议把推理结果也放一下

File: paddlemix/examples/points_qwen2_5/README.md (hunk 1 above)
Comment (zhaop-l) [reply paddlemix/examples/points_qwen2_5/README.md:28]: 我本地测试用的3.0.0, 3.0.0b4没见这个版本的呀？
image

paddlenlp的话本地测试用的就是3.0.0b4

File: paddlemix/demo_images/minicpm_demo.jpeg
Comment (lyuwenyu) [inline paddlemix/demo_images/minicpm_demo.jpeg:1]: 这个图片去掉吧；可以放个链接

File: paddlemix/examples/points_qwen2_5/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+- **预训练数据集过滤**：提出使用困惑度（perplexity）作为指标来过滤预训练数据集。通过这种过滤策略，可以显著减少预训练数据集的规模，同时提升模型的性能。
+
+- **模型融合（Model Soup）**：提出对使用不同视觉指令微调数据集进行微调的模型应用模型融合技术，这可以进一步显著提升模型的性能。
+
+**本仓库支持的模型权重:**
+
+| Model              |
+|--------------------|
+| WePOINTS/POINTS-Qwen-2-5-7B-Chat |
```
Comment (lyuwenyu) [inline paddlemix/examples/points_qwen2_5/README.md:17]: 权重是上传了嘛？

Comment (zhaop-l) [reply paddlemix/examples/points_qwen2_5/README.md:17]: 权重怎么上传呢？哈哈，那个参考文档写的太简略了

**Comment by lyuwenyu** [comment]
这个`推理对齐的标准`可以在这个PR里说明一下嘛

**Comment by zhaop-l** [comment]
由于POINTS-Qwen-2-5-7B-Chat没有整体的forward函数，分别对视觉模块+对齐模块和大语言模型进行对齐测试。测试结果如下：
| 模型                    | image features diff(float32) | llm logits diff (float32) |
| ----------------------- | ---------------------------- | ------------------------- |
| POINTS-Qwen-2-5-7B-Chat | 7.19e-05                     | 6.99e-05                  |

File: paddlemix/examples/points_qwen2_5/README.md (hunk 3 above)
Comment (lyuwenyu) [reply paddlemix/examples/points_qwen2_5/README.md:17]: 哈哈问题不大；这个参数是torch转过来的嘛 可以把转参数的脚步上传一下

Comment (zhaop-l) [reply paddlemix/examples/points_qwen2_5/README.md:17]: 已经按照要求修改

**Review by lyuwenyu — DISMISSED** [review]

File: paddlemix/examples/points_qwen2_5/image_infer.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+# @Time    : 2025/4/19 下午8:37
+# @Author  : zhaop-l(zhaopuzxjc@126.com)
+
+from paddlenlp.transformers import CLIPImageProcessor, Qwen2Tokenizer
+from PIL import Image
+
+from paddlemix.models.points_qwen2_5 import POINTSChatModel
+
+model_path = "WePOINTS/POINTS-Qwen-2-5-7B-Chat"
```
Comment (lyuwenyu) [inline paddlemix/examples/points_qwen2_5/image_infer.py]: 这块是不是应该用转换后的paddle的权重路径；建议使用argparser传入模型路径和图片路径

Comment (zhaop-l) [reply paddlemix/examples/points_qwen2_5/image_infer.py]: 已经按照要求修改了

**Review by lyuwenyu — DISMISSED** [review]

File: paddlemix/examples/points_qwen2_5/README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
+# MiniCPM-V-2_6
```
Comment (lyuwenyu) [inline paddlemix/examples/points_qwen2_5/README.md:1]: 这个名字也改一下？

Comment (lyuwenyu) [inline paddlemix/examples/points_qwen2_5/README.md]: 文档里这个名字也改一下？

Comment (zhaop-l) [reply paddlemix/examples/points_qwen2_5/README.md]: 尬住了，没注意到

**Comment by lyuwenyu** [comment]
lgtm

**Review by lyuwenyu — APPROVED** [review]

**Comment by luotao1** [comment]
hi, @zhaop-l 
- 非常感谢你对飞桨的贡献，我们正在运营一个PFCC组织。PFCC是飞桨开源的贡献者俱乐部，只有给飞桨合入过代码的开发者才能加入，俱乐部里每两周会有一次例会（按兴趣参加），也会时不时办线下meetup面基，详情可见 https://github.com/luotao1 主页说明。
- 如果你对PFCC有兴趣，请发送邮件至 [ext_paddle_oss@baidu.com](mailto:ext_paddle_oss@baidu.com)，我们会邀请你加入~
